File: src-tauri/src/character.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize, Debug)]
pub struct Character {
    pub id: String,
    pub name: String,
    pub game_class: u8,
    pub dailies: CharacterDailies,
    pub weeklies: CharacterWeeklies,
}
#[derive(Deserialize, Serialize, Debug)]
pub struct CharacterDailies {
    pub guild_donation: bool,
    pub chaos_dungeon: bool,
    pub guardian_raid: bool,
    pub una: UnaEntries,
}

#[derive(Deserialize, Serialize, Debug)]
pub struct CharacterWeeklies {
    pub abyss_dungeon: bool,
    pub ship_exchange: bool,
    pub una: UnaEntries,
    pub sylmael: bool,
    pub chaos_dungeon_exchange: bool,
}
#[derive(Deserialize, Serialize, Debug)]
pub struct UnaEntries {
    pub first: bool,
    pub second: bool,
    pub third: bool,
}

File: src/utils/character.ts
import { Character } from "../types/config";
import { config, setConfig } from "../state/config";
import { invoke } from "@tauri-apps/api";
import { v4 as uuid } from "uuid";

interface ICreateCharacter {
    name: string;
    game_class: number;
}
export const createCharacter = (character: ICreateCharacter) => {
    const newCharacter: Character = {
        id: uuid(),
        name: character.name,
        game_class: character.game_class,
        dailies: {
            una: {
                first: false,

                second: false,
                third: false,
            },
            chaos_dungeon: false,
            guardian_raid: false,
            guild_donation: false,
        },
        weeklies: {
            una: {
                first: false,
                second: false,
                third: false,
            },
            sylmael: false,
            abyss_dungeon: false,
            ship_exchange: false,
            chaos_dungeon_exchange: false,
        },
    };
    setConfig({
        ...config(),
        characters: [...config().characters, newCharacter],
    });
    invoke("write_config", { config: config() });
};
